src/UI/CompletionsView.re

```diff
@@ -68,192 +63,186 @@ let completionKindToIcon: option(Ext.CompletionItemKind.t) => int =
     |> Option.value(~default=FontAwesome.question);
   };
 
-let completionKindToColor =
+module Styles = {
+  open Style;
+
+  let outerPosition = (~x, ~y) => [
+    position(`Absolute),
+    top(y - 4),
+    left(x + 4),
+  ];
+
+  let innerPosition = (~height, ~width, ~lineHeight, ~theme: Theme.t) => [
+    position(`Absolute),
+    top(int_of_float(lineHeight +. 0.5)),
+    left(0),
+    Style.width(width),
+    Style.height(height),
+    border(~color=theme.editorSuggestWidgetBorder, ~width=1),
+    backgroundColor(theme.editorSuggestWidgetBackground),
+  ];
+
+  let item = (~isFocused, ~theme: Theme.t) => [
+    isFocused
+      ? backgroundColor(theme.editorSuggestWidgetSelectedBackground)
+      : backgroundColor(theme.editorSuggestWidgetBackground),
+    flexDirection(`Row),
+  ];
+
+  let icon = (~color) => [
+    flexDirection(`Row),
+    justifyContent(`Center),
+    alignItems(`Center),
+    flexGrow(0),
+    backgroundColor(color),
+    width(25),
+  ];
+
+  let label = [flexGrow(1), margin(4)];
+
+  let text =
+      (~highlighted=false, ~theme: Theme.t, ~editorFont: EditorFont.t, ()) => [
+    textOverflow(`Ellipsis),
+    fontFamily(editorFont.fontFile),
+    fontSize(editorFont.fontSize),
+    textWrap(TextWrapping.NoWrap),
+    Style.color(
+      highlighted ? theme.oniNormalModeBackground : theme.editorForeground,
+    ),
+  ];
+
+  let highlightedText = (~theme: Theme.t, ~editorFont: EditorFont.t) =>
+    text(~highlighted=true, ~theme, ~editorFont, ());
+
+  let detail = (~width, ~lineHeight, ~theme: Theme.t) => [
+    position(`Absolute),
+    left(width),
+    top(int_of_float(lineHeight +. 0.5)),
+    Style.width(Constants.maxDetailWidth),
+    flexDirection(`Column),
+    alignItems(`FlexStart),
+    justifyContent(`Center),
+    border(~color=theme.editorSuggestWidgetBorder, ~width=1),
+    backgroundColor(theme.editorSuggestWidgetBackground),
+  ];
+
+  let detailText = (~editorFont: EditorFont.t, ~tokenTheme: TokenTheme.t) => [
+    textOverflow(`Ellipsis),
+    fontFamily(editorFont.fontFile),
+    fontSize(editorFont.fontSize),
+    Style.color(tokenTheme.commentColor),
```

Comment:
Removing the `backgroundColor` actually causes the text rendering quality to degrade - we don't do [gamma correction](https://www.freetype.org/freetype2/docs/text-rendering-general.html) when there is no `backgroundColor` - this causes the text to appear thinned out. Can be a significant degradation depending on the contrast between the background and foreground.

Reply:
Ah, right,I think you've told me this before, but I forget. How hard is it to fix this properly in Revery? It's a real pain to pass the background color around when it's not static. Even worse with the new notifications in the StatusBar where the background and foreground colors are now animated.

Reply:
It is definitely painful, and certainly shouldn't be necessary (and it isn't obvious!)

Skia handles this correctly for us. 

In the meantime, we could keep context of the last 'backgroundColor' we saw in the hierarchy and pass it down, but not sure if it is worth the trouble (since we'd pull it out anyway once Skia is in)

Reply:
Is it worth the trouble to keep passing `backgroundColor` if the only issue is slightly degraded rendering quality, and if Skia will save us soon anyway?

Reply:
I think it is! The un-gamma-corrected text is really bad on some of my monitors, so I'd prefer to pass `backgroundColor` to address it.

> and if Skia will save us soon anyway

I just wish I knew how soon! There are still several blockers to work through before we can integrate it.

Reply:
Alright, will do. I don't think ti's too bad of a fix here. The notifications UI will be more convoluted though.